EIPS/eip-5792.md

```diff
@@ -171,9 +171,9 @@ As with the return value of `wallet_sendCalls`, the batch identifier may be any
 
 ### `wallet_getCapabilities`
 
-An app calling the above-defined methods without prior knowledge that the wallet supports them risks degrading user experience considerably, so an additional RPC for querying the wallet's capabilities is provided.
+An app calling the above-defined methods without prior knowledge that the wallet supports them risks degrading user experience considerably, yet overtly or indirectly querying the wallet for interfaces risks eroding the anonymity of the user-agent (see Privacy Considerations below). The semantics of requesting a capability supports security and privacy by treating unsupported features and unpermitted features the same way, reducing fingerprinting surface. An exemplary wallet RPC is defined below for applications to request the capability to call the RPC methods above, assuming the wallet will process them atomically if the wallet grants that capability.
```

Comment:
> An app calling the above-defined methods without prior knowledge that the wallet supports them risks degrading user experience considerably

Why? The call should fail more or less synchronously and we've defined fallbacks. 

> yet overtly or indirectly querying the wallet for interfaces risks eroding the anonymity of the user-agent (see Privacy Considerations below)

Disagree that this is a substantive concern. 

> An exemplary wallet RPC is defined below for applications to request the capability to call the RPC methods above

Are you referring to wallet_getCapabilities? A wallet could possible implement wallet_sendCalls and not wallet_getCapabilities. It sounds like you're saying `wallet_getCapabilities` is used to detect `wallet_sendCalls` support?

Reply:
"exemplary" means this RPC or equivalent should trump any less-direct way of getting the capability, including discovering the feature without requesting the cap.  you asked for an OCap steelman, so i have to do my best mark miller impersonation until dan f shows up!

Reply:
Does a failure require context switching? Wouldn't the user just never see the notification on their phone or whatever?

Reply:
If a wallet doesn't support an RPC method, then it should synchronously throw a 4200 Unsupported Method error.